src/yolo/yolo.jl

```diff
@@ -374,15 +374,15 @@ mutable struct yolo <: AbstractModel
             end
 
             # precalculate the scale factor from layer-relative to image-relative
-            scale = gpu(reshape(ones(Float32, w*h*2*length(anchormask)*b), w, h, 2, length(anchormask), b))
+            scale = gpu(reshape(similar(out, Float32, w*h*2*length(anchormask)*b), w, h, 2, length(anchormask), b))
```

Comment:
Don't you need to do `scale .= Float32(1)` here? (and below)

though I don't understand why this should be more performant than the specialized `ones`.

Reply:
the `similar` call is to get `out` to be an `AllocArray` (or CUDAArray or whatever), since `ones` always produces `Array`

Reply:
Oh right. Maybe we need a `ones` which can specialize to the array type...